R/list-flatten.R

```diff
@@ -0,0 +1,42 @@
+#' Flatten a list
+#'
+#' Flattening a list removes a single layer of internal hierarchy.
```

Comment:
I would mention something about non-list elements being skipped over.

Part of me wants `list_flatten()` to require that each element of `x` actually be a list, rather than "skipping" non-lists. But I guess that is too strict?

Reply:
> Part of me wants list_flatten() to require that each element of x actually be a list, rather than "skipping" non-lists.

That would be non-standard and make it hard to use with actual trees.

Reply:
Sounds good to me, just a thought